Error reading stream exception thrown from OutputFilter / NativeImageBuildStep

### Describe the bug

Error reading stream exception thrown from OutputFilter / NativeImageBuildStep

We see this issue with Quarkus 2.13

https://github.com/quarkusio/quarkus/blob/main/core/deployment/src/main/java/io/quarkus/deployment/OutputFilter.java was refactored / introduced in https://github.com/quarkusio/quarkus/pull/26892 / Quarkus 2.13.0.Final

Stacktrace:
```
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Building native image from /home/runner/work/quarkus-startstop/quarkus-startstop/app-jax-rs-minimal/target/quarkus-native-image-source-jar/quarkus-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildContainerRunner] Using docker to run the native image builder
Exception in thread "Process stdout" java.lang.RuntimeException: Error reading stream.
	at io.quarkus.deployment.OutputFilter.lambda$apply$0(OutputFilter.java:23)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.IOException: Stream closed
	at java.base/java.io.BufferedInputStream.getBufIfOpen(BufferedInputStream.java:176)
	at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:342)
	at java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
	at java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
	at java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	at java.base/java.io.InputStreamReader.read(InputStreamReader.java:181)
	at java.base/java.io.BufferedReader.fill(BufferedReader.java:161)
	at java.base/java.io.BufferedReader.readLine(BufferedReader.java:326)
	at java.base/java.io.BufferedReader.readLine(BufferedReader.java:392)
	at io.quarkus.deployment.OutputFilter.lambda$apply$0(OutputFilter.java:19)
	... 1 more
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildContainerRunner] Checking image status quay.io/quarkus/ubi-quarkus-native-image:22.2-java17
```

Example logs: https://github.com/quarkus-qe/quarkus-startstop/actions/runs/3313239997 - see `ci-artifacts` and explore the directory structure of the zipped content. Alternatively click on `Linux - Native build` to see which test failed.

Issues is not regularly reproducible, we see it in daily GitHub action.

https://github.com/quarkus-qe/quarkus-startstop/actions/workflows/ci.yaml
<img width="792" alt="Screenshot 2022-10-24 at 20 08 28" src="https://user-images.githubusercontent.com/925259/197595561-e0403543-8bbc-4cd1-ad93-bb30f8869d15.png" >



### Expected behavior

No error

### Actual behavior

~50% of StartStopTS runs hit at least one "Exception in thread "Process stdout" java.lang.RuntimeException: Error reading stream." during native build executions. There are 8 compilations into native binary in one TS run.

### How to Reproduce?

https://github.com/quarkus-qe/quarkus-startstop/actions/workflows/ci.yaml

https://github.com/quarkus-qe/quarkus-startstop/blob/main/.github/workflows/ci.yaml#L60

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

Java 11

### GraalVM version (if different from Java)

ubi-quarkus-native-image:22.2-java17

### Quarkus version or git rev

2.13.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/core/deployment/src/main/java/io/quarkus/deployment/OutputFilter.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/OutputFilter.java
@@ -6,8 +6,11 @@ import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.util.function.Function;
 
+import org.jboss.logging.Logger;
+
 public class OutputFilter implements Function<InputStream, Runnable> {
     private final StringBuilder builder = new StringBuilder();
+    private static final Logger log = Logger.getLogger(OutputFilter.class);
 
     @Override
     public Runnable apply(InputStream is) {
@@ -20,7 +23,11 @@ public class OutputFilter implements Function<InputStream, Runnable> {
                     builder.append(line);
                 }
             } catch (IOException e) {
-                throw new RuntimeException("Error reading stream.", e);
+                if (e.getMessage().contains("Stream closed")) {
+                    log.warn("Stream is closed, ignoring and trying to continue");
+                } else {
+                    throw new RuntimeException("Error reading stream.", e);
+                }
             }
         };
     }

